components/Tabs/Tabs.js

```diff
@@ -1,62 +1,83 @@
-    // this.data;
-    
+     this.data = this.link.dataset.tab;
+    //console.log(this.data);
+
     // Using the custom data attribute get the associated Item element
-    // this.itemElement;
-    
+
+     this.itemElement = document.querySelector(`.tabs-item[data-tab="${tab, this.data}"]`);
+     //console.log(this.itemElement)
+
     // Using the Item element, create a new instance of the TabItem class
-    // this.tabItem;
-    
+     this.tabItem = new TabItem(tab, this.itemElement);
+      //console.log(this.tabItem)
     // Add a click event listener on this instance, calling the select method on click
-
+     this.link.addEventListener('click', this.select.bind(this));
   };
 
   select() {
     // Get all of the elements with the tabs-link class
-    // const links;
+     const links = document.querySelectorAll('.tabs-link');
 
     // Using a loop or the forEach method remove the 'tabs-link-selected' class from all of the links
-    // Array.from(links).forEach();
+     Array.from(links).forEach(link => {
+
+        link.classList.remove('tabs-link-selected');
+      // console.log('testing')
+     });
 
     // Add a class named "tabs-link-selected" to this link
-    // this.element;
-    
-    // Call the select method on the item associated with this link
+     this.link.classList.toggle('tabs-link-selected');
 
+    // Call the select method on the item associated with this link
+     this.tabItem.select();
   }
 }
 
 class TabItem {
-  constructor(element) {
+  constructor(tab, data) {
     // Assign this.element to the passed in element
-    // this.element;
+     this.element = data;
+     this.tab = tab;
+
   }
 
   select() {
     // Select all ".tabs-item" elements from the DOM
-    // const items;
+    const items = Array.from(document.querySelectorAll('.tabs-item'));
 
     // Remove the class "tabs-item-selected" from each element
-    
+     items.forEach(item => {
+       item.classList.remove('tabs-item-selected');
+     })
+
     // Add a class named "tabs-item-selected" to this element
-    //this.element;
+     this.element.classList.toggle('tabs-item-selected');
   }
 }
 
-/* START HERE: 
-
+/* START HERE:
+//
 - Select all classes named ".tabs-link" and assign that value to the links variable
 
 - With your selection in place, now chain a .forEach() method onto the links variable to iterate over the DOM NodeList
 
 - In your .forEach() method's callback function, return a new instance of TabLink and pass in each link as a parameter
 
 */
+// const items = document.querySelectorAll('.tabs-item').forEach(item => {const leedle = new TabItem(item)});
+// let links = document.querySelectorAll('.tabs-link').forEach(link => new TabLink(link));
 
-links = document.querySelectorAll();
\ No newline at end of file
+const tabs = document.querySelectorAll('.tabs');
+tabs.forEach(tab => {
+  const links = tab.querySelectorAll('.tabs-link');
+  //console.log('Navigator:', links.length)
```

Comment:
I noticed that these lines are part of the changes in the commit you labeled as a "`Tabs` constructor function". However, for clarification, this isn't constructing a new component; what is actually happening is that you grabbed all the container elements for groups of tabs (everything with `class="tabs"`) and then instantiated all the tab link elements inside these containers.